hist/histpainter/src/TGraphPainter.cxx

```diff
@@ -4429,7 +4430,9 @@ void TGraphPainter::PaintScatter(TScatter *theScatter, Option_t* chopt)
          } else {
             c = theColor[i];
          }
-         theScatter->SetMarkerColor(gStyle->GetColorPalette(((c-minc)/(maxc-minc))*nbcol));
+         nc = ((c-minc)/(maxc-minc))*nbcol;
```

Comment:
May be better use double -> int conversion here like:
```
nc = TMath::Nint(((c-minc)/(maxc-minc))*nbcol);
```
I expect warnings one some compilers.

Reply:
And probably arithmetic should be:
```
... * (nbcol - 1);
```